test/media-segment-request.test.js

```diff
@@ -193,6 +193,50 @@ QUnit.test('cancels outstanding key requests on timeout', function(assert) {
   this.clock.tick(2000);
 });
 
+QUnit.test('does not wait for other requests to finish when one request errors',
+function(assert) {
+  let keyReq;
+  let abortedKeyReq = false;
+  const done = assert.async();
+
+  assert.expect(8);
+  mediaSegmentRequest(
+    this.xhr,
+    this.xhrOptions,
+    this.noop,
+    this.noop,
+    {
+      resolvedUri: '0-test.ts',
+      key: {
+        resolvedUri: '0-key.php'
+      }
+    },
+    this.noop,
+    (error, segmentData) => {
+      assert.notOk(keyReq.aborted, 'did not run original abort function');
+      assert.ok(abortedKeyReq, 'ran overridden abort function');
+      assert.equal(error.code, REQUEST_ERRORS.FAILURE, 'request failed');
+
+      done();
+    });
+  assert.equal(this.requests.length, 2, 'there are two requests');
+
+  keyReq = this.requests.shift();
+  // Typically, an abort will run the error algorithm for an XHR, however, in certain
+  // cases (e.g., if the request is unsent), the error algorithm will not be run and
+  // the request will never "finish." In order to mimic this behavior, override the
+  // default abort function so that it doesn't finish.
+  keyReq.abort = () => {
+    abortedKeyReq = true;
```

Comment:
might want to rename `abortedKeyReq` so it's clear that we're overriding default behavior

Reply:
Alternatively, can we mimic this scenario by setting the XHR state to unsent?

Reply:
I just tried that, and we can, the only thing I was wary of after I wrote it up is I still wanted to have a way to verify that we called abort on the key request, otherwise, since we mock other aspects of the XHR, I'd worry that the state would always be in unsent for some reason and we didn't actually call abort properly. Though maybe that would cause problems elsewhere. What do you think?

Reply:
I think this makes sense for the purpose of verifying we called abort. Can we also verify that the XHR state changed in the test or does the mocking make that difficult/not useful?

Reply:
The XHR state for the key request optimally shouldn't actually change if we want to reflect the browser behavior (that was the cause of the bug, that it remains in its original, unsent, state), but with the mocking and setup, the state does change (why we override the abort in this case).